paymentsheet/src/main/java/com/stripe/android/paymentsheet/ui/WalletButtonsInteractor.kt

```diff
@@ -0,0 +1,187 @@
+package com.stripe.android.paymentsheet.ui
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Immutable
+import androidx.lifecycle.viewModelScope
+import com.stripe.android.CardBrandFilter
+import com.stripe.android.common.model.CommonConfiguration
+import com.stripe.android.link.ui.LinkButton
+import com.stripe.android.lpmfoundations.paymentmethod.PaymentMethodMetadata
+import com.stripe.android.lpmfoundations.paymentmethod.PaymentSheetCardBrandFilter
+import com.stripe.android.paymentelement.confirmation.ConfirmationHandler
+import com.stripe.android.paymentelement.confirmation.toConfirmationOption
+import com.stripe.android.paymentsheet.PaymentSheet
+import com.stripe.android.paymentsheet.flowcontroller.FlowControllerViewModel
+import com.stripe.android.paymentsheet.model.PaymentSelection
+import com.stripe.android.paymentsheet.state.PaymentElementLoader
+import com.stripe.android.paymentsheet.utils.asGooglePayButtonType
+import com.stripe.android.uicore.utils.combineAsStateFlow
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+
+internal interface WalletButtonsInteractor {
+    val state: StateFlow<State>
+
+    class State(
+        val walletButtons: List<WalletButton>,
+        val buttonsEnabled: Boolean,
+    )
+
+    interface WalletButton {
```

Comment:
This pattern seems to combine both state + UI in the same object.

It's useful to have them separate, and keep the UI as total UI code for unit testing purposes.